chaos/chaos-server.go

```diff
@@ -47,14 +50,42 @@ func startRPCServer(port int) {
 		lockMap: make(map[string][]lockRequesterInfo),
 		// timestamp: leave uninitialized for testing (set to real timestamp for actual usage)
 	}
-	go func() {
+	lockHeartBeat := uint64(0)
+
+	lockMaintainer := func() {
 		// Start with random sleep time, so as to avoid "synchronous checks" between servers
 		time.Sleep(time.Duration(rand.Float64() * float64(LockMaintenanceLoop)))
 		for {
 			time.Sleep(LockMaintenanceLoop)
 			locker.lockMaintenance(LockCheckValidityInterval)
+			atomic.AddUint64(&lockHeartBeat, 1)
 		}
-	}()
+	}
+	lockMonitor := func() {
+		lastHeartBeat := uint64(0)
+		lastHeartBeatStamp := time.Now()
+		for {
+			time.Sleep(LockMaintenanceLoop)
+			currentHeartBeat := atomic.LoadUint64(&lockHeartBeat)
+			if currentHeartBeat > lastHeartBeat {
+				lastHeartBeat = currentHeartBeat
+				lastHeartBeatStamp = time.Now()
+			}
+			// Check whether not too much time has passed since last 'sign' of life
+			// from maintenance loop
+			if time.Since(lastHeartBeatStamp) > LockMaintenanceHealthThreshold {
```

Comment:
In `lockMaintainer()`, `lockHeartBeat` is incremented after max time of `LockMaintenanceLoop` + `LockCheckValidityInterval` which is < `LockMaintenanceHealthThreshold`, that means this `if` won't be executed at all.  Could you clarify whether I am missing something?